Create snapshot.md (#290)
gridcoin-community/Gridcoin-Site: wiki/snapshot.md

---
title: Snapshots
layout: wiki
---

# Overview

> Snapshots are a snapshot of the chain at a particular blockheight.

Reviewer: The sentence may be a bit technical for the first sentence. Perhaps something like this would be better
```suggestion
Snapshots are a copy of the blockchain at a particular point in time.
```

> Gridcoin snapshots are created by trusted developers and they allow newer users to get into sync faster than syncing from 0.

Reviewer: Snapshots being created by trusted developers is not necessarily true. Anyone can create a snapshot if they want to. The snapshots on `snapshot.gridcoin.us` are probably what you are referring to, but I think it may just be better to remove that part of the sentence

```suggestion
Snapshots can help newer users to get into sync faster than syncing from 0.
```

# Issues
* Snapshots are not validated by the wallet.
* Snapshots are an inherently centralised method of getting into sync.
* Users may mistakenly install a malicious or corrupted snapshot. 
If enough users install the bad snapshot then the integrity of the chain could be compromised.

# Using snapshots

> ## Starting from 5.0.0

Reviewer: Since people aren't (or at least shouldn't be) running <5.0.0.0 anymore, I think we can just say something like "In the Wallet" instead 
```suggestion
## In the Wallet
```

Snapshots can now be downloaded and applied automatically by the wallet.

> It is accessible from File -> Snapshot download

Reviewer: ```suggestion
It is accessible from File -> Snapshot Download or the `-snapshotdownload` [startup option](cmd-options "wikilink")
```

## Manual method
1. [Download the snapshot](https://snapshot.gridcoin.us/snapshot.zip)
2. [Verify the checksum (IMPORTANT)](https://snapshot.gridcoin.us/snapshot.zip.sha256)
3. Delete the following folders: `txleveldb` & `accrual` from your Gridcoin data directory.
4. Delete the `blk****.dat` files from your Gridcoin data directory.
5. Extract the snapshot zip into your Gridcoin data directory.
6. Restart the wallet and get into sync.

> # Alternatives

Reviewer: I might add something here about how to create your own snapshots for yourself instead of relying on an external one

* Syncing from 0:
  
  This is preferred. With all of the recent changes in the wallet it is only mildly slower than using a snapshot.
  This doesn't require trust in the snapshot creator.
* Bootstrap:
  
  As of April 2021 these are no longer offered.
  It doesn't bypass the checks in the wallet which prevent malicious bootstraps from damaging the chain.
  There is little to no point in using one of these over syncing from 0. 

>   It doesn't inherently bring any speed benefits compared to syncing normally but requires on a centralised download.

Reviewer: I believe it can technically help speed things up, but yeah not by much
```suggestion
  It brings limited speed benefits compared to syncing normally but requires on a centralised download.
```

---
title: Snapshots
layout: wiki
---

# Overview
Snapshots are a copy of the blockchain at a particular point in time.
Snapshots can help newer users to get into sync faster than syncing from 0.

# Issues
* Snapshots are not validated by the wallet.
* Snapshots are an inherently centralised method of getting into sync.
* Users may mistakenly install a malicious or corrupted snapshot. 
If enough users install the bad snapshot then the integrity of the chain could be compromised.

# Using snapshots
## In the Wallet
Snapshots can now be downloaded and applied automatically by the wallet.

It is accessible from File -> Snapshot Download or the `-snapshotdownload` [startup option](cmd-options "wikilink")

> ## Manual method

Reviewer: Should mention that you need to shutdown the wallet here as well

> 1. [Download the snapshot](https://snapshot.gridcoin.us/snapshot.zip)

Reviewer: Just realized we should add a step that says "Shutdown the wallet if running"

2. [Verify the checksum (IMPORTANT)](https://snapshot.gridcoin.us/snapshot.zip.sha256)
3. Delete the following folders: `txleveldb` & `accrual` from your Gridcoin data directory.
4. Delete the `blk****.dat` files from your Gridcoin data directory.
5. Extract the snapshot zip into your Gridcoin data directory.
6. Restart the wallet and get into sync.

## Creating your own snapshot
1. Shutdown your wallet.
2. Create an empty archive in an archive program of your choice.
3. Add all the `blk****.dat` files to your archive.
4. Add the `txleveldb` folder to your archive.

> 5. Follow the manual method on your other machine.

Reviewer: Might be good to clarify that you don't use it immediatelyand that you use your own archive instead of the snapshot.gridcoin.us one 
```suggestion
5. When you want to use it, follow the manual method but with your archive```
